File: src/test/java/br/com/zumbolovsky/fateapp/MongoDBSetup.java
package br.com.zumbolovsky.fateapp;

import org.testcontainers.containers.MongoDBContainer;

public class MongoDBSetup {
    private static final String image = "mongo:latest";
    private static final Integer port = 27017;

    private static MongoDBContainer container() {
        final MongoDBContainer mongoDBContainer = new MongoDBContainer(image);
        mongoDBContainer.withExposedPorts(port);
        return mongoDBContainer;
    }

    private String database() {
        return "test";
    }

    public MongoDBSetup() {
        final MongoDBContainer container = container();
        container.start();
        System.setProperty("spring.data.mongodb.uri", container.getReplicaSetUrl());
        System.setProperty("spring.data.mongodb.database", database());
    }
}

File: src/test/kotlin/br/com/zumbolovsky/fateapp/MongoDBSetup.kt
package br.com.zumbolovsky.fateapp

import org.testcontainers.containers.MongoDBContainer

class MongoDBSetup {
    private val image = "mongo:latest"
    private val port = 27017

    private val container =
        MongoDBContainer(image)
            .withExposedPorts(port)

    private val uri: String
    get() = container.replicaSetUrl

    private val database: String
    get() = "test"

    init {
        container.start()
        System.setProperty("spring.data.mongodb.uri", uri)
        System.setProperty("spring.data.mongodb.database", database)
    }
}
